### PR TITLE
VCITA2-411 - add swagger for coupons

### PR DESCRIPTION
**Pull request by VCeti**
Added swagger for sales/coupons api

### CODE REVIEW

File: swagger/sales/coupon.json [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+              "description": "Get a list of Coupons - Available for **Business Tokens**",
+              "parameters": [
+                {
+                  "name": "filter",
+                  "in": "query",
+                  "schema": {  
+                    "type": "string" 
+                  },
+                  "description": "Filter parameter filter[by_status] = active / expiered / scheduled / still_available",
+                  "required": false
```
Comment (ramsterita) [inline swagger/sales/coupon.json:64]: we do not support filter anymore. if you enable filtering please state them explicitly

File: swagger/sales/coupon.json [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  "servers": [
+      {
+          "url": "https://api.vcita.biz",
+          "description": "API Gateway server"
+      }
+  ],
+  "components": {
+      "schemas": {
+          "Response": {
+              "type": "object",
```
Comment (ramsterita) [inline swagger/sales/coupon.json:19]: I don't think you need this. its not used

File: swagger/sales/coupon.json [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                      "type": "object"
+                  }
+              },
+              "required": [
+                  "success",
+                  "data"
+              ]
+          },
+          "Object": {
+              "type": "object",
```
Comment (ramsterita) [inline swagger/sales/coupon.json:34]: I don't think you need this, its not used

File: swagger/sales/coupon.json [hunk 4]
```diff
@@ -0,0 +1,124 @@
+{
+  "openapi": "3.0.0",
+  "info": {
+      "title": "Sales",
+      "description": "An set of endpoints to manage Coupons and other sales data in the inTandem platform",
+      "version": "3.0",
+      "contact": {}
```
Comment (ramsterita) [inline swagger/sales/coupon.json:7]: no need for contact and tags here

**Review by ramsterita — CHANGES_REQUESTED** [review]

File: swagger/sales/coupon.json (hunk 1 above)
Comment (VCeti) [reply swagger/sales/coupon.json:64]: Do you mean I need a parameter for by_status?

File: entities/sales/Coupon.json [hunk 5]
```diff
@@ -6,78 +6,78 @@
...
       "type": "string",
       "enum": [
         "fixed", "percent"
       ],
-      "description": "Type of discount offered, fixed: for fixed amount, percent: for percentage disscount."
+      "description": "Type of discount offered, fixed: for fixed amount, percent: for percentage disscount"
     },
     "discount_value": {
       "type": "number",
-      "description": "The value of the discount."
```
[the remark this answers is not in the document]
Comment (VCeti) [reply entities/sales/Coupon.json:24]: For example : 5 $ or 5%

[the remark this answers is not in the document]
Comment (ramsterita) [reply entities/sales/Coupon.json:24]: So add it to the description:
The discount amount give by applying the coupon. If the coupon type is "fixed" then the amount is decimal ($5.5), keeping the same currency as the item. If the type is "percentage" then the amount is decimal of percentage (5.3%).

**Pull request merged**